src/Storages/MergeTree/RequestResponse.cpp

```diff
@@ -122,6 +123,7 @@ void InitialAllRangesAnnouncement::deserialize(ReadBuffer & in)
             "from replicas differ. Got: {}, supported version: {}",
             version, DBMS_PARALLEL_REPLICAS_PROTOCOL_VERSION);
 
+    readIntBinary(mode, in);
```

Comment:
1. Let's validate that  `mode` is correct.
2. Let's for `CoordinationMode` enum specify values like `Default=0, WithOrder=1, ReverseOrder=2`. And write a comment that they are serialized, so that you can't reorder it or add another mode in the middle.